File: standardnaast-backend/src/main/java/standardNaast/backend/service/MemberService.java
package standardNaast.backend.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import standardNaast.backend.dto.MemberCreateUpdateDto;
import standardNaast.backend.dto.MemberDto;

import java.util.Optional;

public interface MemberService {

    Page<MemberDto> getMembers(String search, Pageable pageable);

    MemberDto getMemberById(Long id);

    Optional<MemberDto> getMemberByMemberNumber(Long memberNumber);

    MemberDto createMember(MemberCreateUpdateDto dto);

    MemberDto updateMember(Long id, MemberCreateUpdateDto dto);

    void deleteMember(Long id);

    Long getNextMemberNumber();
}

File: standardnaast-ng/src/app/core/services/member.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Member, MemberCreateUpdate } from '../models/member.model';
import { Page } from '../models/page.model';

@Injectable({
  providedIn: 'root'
})
export class MemberService {
  private readonly apiUrl = `${environment.apiUrl}/members`;

  constructor(private http: HttpClient) {}

  getMembers(search?: string, page = 0, size = 20, sort = 'memberNumber,asc'): Observable<Page<Member>> {
    let params = new HttpParams()
      .set('page', page.toString())
      .set('size', size.toString())
      .set('sort', sort);

    if (search && search.trim()) {
      params = params.set('search', search.trim());
    }

    return this.http.get<Page<Member>>(this.apiUrl, { params });
  }

  getMemberById(id: number): Observable<Member> {
    return this.http.get<Member>(`${this.apiUrl}/${id}`);
  }

  createMember(member: MemberCreateUpdate): Observable<Member> {
    return this.http.post<Member>(this.apiUrl, member);
  }

  updateMember(id: number, member: MemberCreateUpdate): Observable<Member> {
    return this.http.put<Member>(`${this.apiUrl}/${id}`, member);
  }

  deleteMember(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  getNextMemberNumber(): Observable<number> {
    return this.http.get<number>(`${this.apiUrl}/next-number`);
  }
}
